frontend/views/containers/GroupPledgesGraph.vue

```diff
@@ -1,218 +1,195 @@
+      'ourContractId'
+    ]),
     graphData () {
-      const { userPledgeAmount, userIncomeAmount } = this
-      const { groupMembersTotal, groupPledgeGoal, mincome, othersPledges } = this.fakeStore
-      const othersAmount = othersPledges.reduce((acc, cur) => acc + cur, 0)
-      const userIncomeNeeded = userIncomeAmount !== null && userIncomeAmount < mincome ? mincome - userIncomeAmount : null
-      const totalAmount = othersAmount + userPledgeAmount
-      const goal = groupPledgeGoal + userIncomeNeeded
-      const members = othersPledges.length + (userPledgeAmount ? 1 : 0)
-      const neededPledges = goal - totalAmount
-      const surplus = totalAmount - groupPledgeGoal
-      const membersNeedingPledges = groupMembersTotal - othersPledges.length
-
-      // Dumb Rule of 3 algorithm to know how much the user will receive (userIncomeToReceive) - just for layout purposes.
-      const avgMembersNeedingPledges = goal / membersNeedingPledges
-      const avgNeededPledges = neededPledges / membersNeedingPledges
-      const userIncomeToReceive = userIncomeNeeded ? (userIncomeNeeded - (userIncomeNeeded * avgNeededPledges / avgMembersNeedingPledges)).toFixed(2) : 0
+      const mincome = this.groupSettings.mincomeAmount
+      const ourPledgeAmount = this.type === 'pledgeAmount' && this.amount >= 0 && this.amount
+      const ourIncomeAmount = this.type === 'incomeAmount' && this.amount < mincome && this.amount
+      const doWeNeedPledge = typeof ourIncomeAmount === 'number'
+      let othersIncomeNeeded = 0
+      let othersPledgesAmount = 0
+      let othersPledgesCount = 0
+
+      Object.keys(this.groupProfiles).forEach(username => {
+        const { incomeDetailsType, contractID, ...profile } = this.groupProfiles[username]
+
+        if (contractID === this.ourContractId) { return }
+
+        const amount = profile[incomeDetailsType]
+
+        if (incomeDetailsType === 'incomeAmount') {
+          othersIncomeNeeded += mincome - amount
+        } else if (incomeDetailsType === 'pledgeAmount') {
+          othersPledgesAmount += amount
+          othersPledgesCount += 1
+        }
+      })
+
+      const ourIncomeNeeded = doWeNeedPledge ? mincome - ourIncomeAmount : null
+      const pledgeTotal = othersPledgesAmount + ourPledgeAmount
+      const groupGoal = othersIncomeNeeded + ourIncomeNeeded
+      const neededPledges = groupGoal - pledgeTotal
+      const surplus = pledgeTotal - othersIncomeNeeded - ourIncomeNeeded
+
+      let ourIncomeToReceive
+      if (neededPledges > 0) {
+        // TODO - get real income to receive based on distribution algorithm
+        const membersNeedingPledges = this.groupMembersCount - othersPledgesCount
+        const avgNeededPledges = neededPledges / membersNeedingPledges
+        const avgMembersNeedingPledges = groupGoal / membersNeedingPledges
+        ourIncomeToReceive = +(ourIncomeNeeded - (ourIncomeNeeded * avgNeededPledges / avgMembersNeedingPledges)).toFixed(0)
```

Comment:
Question: @taoeffect can u suggest a clean/easy way of using `IncomeDistribution` here without repeating the code in the getter `groupIncomeDistribution`?

Reply:
I'm taking it easy today, but for when I get back, can you let me know what you mean by "using `IncomeDistribution`"? I don't see that defined anywhere.

Reply:
IncomeDistribution, I refer to the following file imported at `state.js`, used to create the getter `groupIncomeDistribution`:

> import incomeDistribution from '~/frontend/utils/distribution/mincome-proportional.js'

Reply:
OK thanks! I'll have a close look at what you're doing here when after the weekend (or perhaps tomorrow, we'll see)

Reply:
OK, yes, so here's what needs to be done:

1. Up above in the loop starting at line 82, convert that to a for/in loop that looks similar to the loop in `groupIncomeDistribution` in order to fill up an array called `newIncomeDistribution` (just like the `currentIncomeDistribution` inside of `groupIncomeDistribution`)
2. Outside of the loop, perhaps in here, push again to newIncomeDistribution `{ name: username, amount: ourIncomeAmount }`
3. In here, do something like `ourIncomeToReceive = incomeDistribution(newIncomeDistribution, mincome).filter(i => i.to === username).reduce((acc, cur) => cur.amount + acc, 0)` (double-check me there) to generate `ourIncomeToReceive`